lib/link_state.h

```diff
@@ -0,0 +1,780 @@
+#define LS_ATTR_NAME		0x00000001
+#define LS_ATTR_METRIC		0x00000002
+#define LS_ATTR_TE_METRIC	0x00000004
+#define LS_ATTR_ADM_GRP		0x00000008
+#define LS_ATTR_LOCAL_ADDR	0x00000010
+#define LS_ATTR_NEIGH_ADDR	0x00000020
+#define LS_ATTR_LOCAL_ADDR6	0x00000040
+#define LS_ATTR_NEIGH_ADDR6	0x00000080
+#define LS_ATTR_LOCAL_ID	0x00000100
+#define LS_ATTR_NEIGH_ID	0x00000200
+#define LS_ATTR_MAX_BW		0x00000400
+#define LS_ATTR_MAX_RSV_BW	0x00000800
+#define LS_ATTR_UNRSV_BW	0x00001000
+#define LS_ATTR_REMOTE_AS	0x00002000
+#define LS_ATTR_REMOTE_ADDR	0x00004000
+#define LS_ATTR_REMOTE_ADDR6	0x00008000
+#define LS_ATTR_DELAY		0x00010000
+#define LS_ATTR_MIN_MAX_DELAY	0x00020000
+#define LS_ATTR_JITTER		0x00040000
+#define LS_ATTR_PACKET_LOSS	0x00080000
+#define LS_ATTR_AVA_BW		0x00100000
+#define LS_ATTR_RSV_BW		0x00200000
+#define LS_ATTR_USE_BW		0x00400000
+#define LS_ATTR_ADJ_SID		0x00800000
+#define LS_ATTR_BCK_ADJ_SID	0x01000000
+#define LS_ATTR_SRLG		0x02000000
+
+/* Link State Attributes */
+struct ls_attributes {
+	uint32_t flags;			/* Flag for parameters validity */
+	struct ls_node_id adv;		/* Adv. Router of this Link State */
+	char name[MAX_NAME_LENGTH];	/* Name of the Edge. Could be null */
+	struct {			/* Standard TE metrics */
+		uint32_t metric;		/* IGP standard metric */
+		uint32_t te_metric;		/* Traffic Engineering metric */
+		uint32_t admin_group;		/* Administrative Group */
+		struct in_addr local;		/* Local IPv4 address */
+		struct in_addr remote;		/* Remote IPv4 address */
+		struct in6_addr local6;		/* Local IPv6 address */
+		struct in6_addr remote6;	/* Remote IPv6 address */
+		uint32_t local_id;		/* Local Identifier */
+		uint32_t remote_id;		/* Remote Identifier */
+		float max_bw;			/* Maximum Link Bandwidth */
+		float max_rsv_bw;		/* Maximum Reservable BW */
+		float unrsv_bw[8];		/* Unreserved BW per CT (8) */
+		uint32_t remote_as;		/* Remote AS number */
+		struct in_addr remote_addr;	/* Remote IPv4 address */
+		struct in6_addr remote_addr6;	/* Remote IPv6 address */
+	} standard;
+#define LS_ATTR_STANDARD_SIZE	124
+	struct {		/* Extended TE Metrics */
+		uint32_t delay;		/* Unidirectional average delay */
+		uint32_t min_delay;	/* Unidirectional minimum delay */
+		uint32_t max_delay;	/* Unidirectional maximum delay */
+		uint32_t jitter;	/* Unidirectional delay variation */
+		uint32_t pkt_loss;	/* Unidirectional packet loss */
+		float ava_bw;		/* Available Bandwidth */
+		float rsv_bw;		/* Reserved Bandwidth */
+		float used_bw;		/* Utilized Bandwidth */
+	} extended;
+#define LS_ATTR_EXTENDED_SIZE	32
+	struct {		/* (LAN)-Adjacency SID for OSPF */
+		uint32_t sid;		/* SID as MPLS label or index */
+		uint8_t flags;		/* Flags */
+		uint8_t weight;		/* Administrative weight */
+		union {
+			struct in_addr addr;	/* Neighbor @IP for OSPF */
+			uint8_t sysid[ISO_SYS_ID_LEN]; /* or Sys-ID for ISIS */
+		} neighbor;
+	} adj_sid[2];		/* Primary & Backup (LAN)-Adj. SID */
+#define LS_ATTR_ADJ_SID_SIZE	12o
```

Comment:
Syntax in 12o , should be 120

Reply:
Good catch.

I corrected the error and push force again. Hope CI/CD will not failed